### PR TITLE
Deprecate runtime specific base images

### PR DESCRIPTION
**Pull request by andreaTP**
### What changes were proposed in this pull request?
Deprecate the usage of runtime-specific base images in favor of giving full control to the user over the generation of the streamlets images.

### Why are the changes needed?
To have a self-contained build of the streamlet's images and get rid of the separate generation of the runtime base images.

### Does this PR introduce any user-facing change?
Yes in the sbt plugin:

 - `cloudflowDockerParentImage` (previously unused) has been renamed `cloudflowDockerBaseImage` and used to store the docker image we use as `from`
 - `cloudflowFlinkBaseImage`, `cloudflowSparkBaseImage` and `cloudflowAkkaBaseImage` are now deprecated, this means that users that are using them will not benefit from this change and they should migrate to `cloudflowDockerBaseImage`
 - the first time users will build and push they are going to miss caches and it will take some time

### How was this patch tested?
adding more sbt scripted tests, running the integration tests and `taxi-ride` with this change in.

### CODE REVIEW

File: core/sbt-cloudflow/src/main/resources/runtimes/flink/config.sh [hunk 1]
```diff
@@ -405,21 +405,9 @@ if [ -z "${FLINK_PID_DIR}" ]; then
     FLINK_PID_DIR=$(readFromConfig ${KEY_ENV_PID_DIR} "${DEFAULT_ENV_PID_DIR}" "${YAML_CONF}")
 fi
 
+if [ -z "${FLINK_ENV_JAVA_OPTS}" ]; then
```
Comment (RayRoestenburg) [inline core/sbt-cloudflow/src/main/resources/runtimes/flink/config.sh:408]: @franciscolopezsancho

File: core/sbt-cloudflow/src/main/resources/runtimes/flink/flink-console.sh [hunk 2]
```diff
@@ -70,4 +70,4 @@ if [[ ${JAVA_VERSION} =~ ${IS_NUMBER} ]]; then
 fi
 
 echo "Starting $SERVICE as a console application on host $HOSTNAME."
-exec $JAVA_RUN $JVM_ARGS "${log_setting[@]}" ${FLINK_ENV_JAVA_OPTS} -classpath "`manglePathList "$FLINK_TM_CLASSPATH:$INTERNAL_HADOOP_CLASSPATHS"`" ${CLASS_TO_RUN} "${ARGS[@]}"
```
Comment (RayRoestenburg) [inline core/sbt-cloudflow/src/main/resources/runtimes/flink/flink-console.sh:73]: @franciscolopezsancho

File: core/sbt-cloudflow/src/main/resources/runtimes/flink/flink-entrypoint.sh [hunk 3]
```diff
@@ -32,11 +32,6 @@ drop_privs_cmd() {
 }
 
 # Add in extra configs set by the operator
-# If not already present
-if  ! cat "$FLINK_HOME/conf/flink-conf.yaml" | grep -q "taskmanager.memory.flink.size" ; then
```
Comment (RayRoestenburg) [inline core/sbt-cloudflow/src/main/resources/runtimes/flink/flink-entrypoint.sh:36]: This should go back in

File: core/sbt-cloudflow/src/main/resources/runtimes/flink/flink-console.sh (hunk 2 above)
Comment (franciscolopezsancho) [reply core/sbt-cloudflow/src/main/resources/runtimes/flink/flink-console.sh:73]: Well spotted @RayRoestenburg , this was in that order to allow the user to set logging options through conf. If `${FLINK_ENV_JAVA_OPTS}` goes after  `${log_setting[@]}` no override of logging would be possible.

**Pull request merged**